src/Emulator.cpp

```diff
@@ -37,21 +32,26 @@ void Emulator::Start()
                 }
             }
 
+            if (!m_isRunning)
+            {
+                // Exit early if the app is closing
+                continue;
+            }
+
             // Emulate one frame on the CPU (70244 cycles)
             m_cpu->StepFrame();
+            Render();
 
-            // The frame is over when we have spent 16ms here
-            if(m_timePerFrame - frameTime > 0)
+            // If we haven't used up our time, we need to delay the rest of the frame time
+            Uint32 frameElapsedTime = SDL_GetTicks() - frameStartTime;
+            if (frameElapsedTime < TimePerFrame)
```

Comment:
Doing it this way prevents the underflow. FYI: This is a very common mistake, and a lot of interviewers will actually ask this question.  In short, instead of DOING the subtract, compare without the add/subtract. Then react.

In this case, if the "frameElapsedTime" is less than the "TimePerFrame", then we need to delay for the difference. And we KNOW TimePerFrame - frameElapsedTime will NEVER be less than zero!